### PR TITLE
Moving more documentation to use codegen

### PR DESCRIPTION
**Pull request by alexanderbock**

### CODE REVIEW

**Comment by alexanderbock** [comment]
Not yet ready

**Comment by alexanderbock** [comment]
Ready for some 👀 now

File: modules/spacecraftinstruments/rendering/renderablefov.cpp [hunk 1]
```diff
@@ -342,61 +248,37 @@ RenderableFov::RenderableFov(const ghoul::Dictionary& dictionary)
...
-            StandoffDistanceInfo.identifier
-        ));
-    }
+    _lineWidth = p.lineWidth.value_or(_lineWidth);
+    addProperty(_lineWidth);
 
-    if (dictionary.hasKey(KeyBoundsSimplification)) {
-        _simplifyBounds = dictionary.value<bool>(KeyBoundsSimplification);
-    }
+    _standOffDistance = p.standOffDistance.value_or(_lineWidth);
```
Comment (WeirdRubberDuck) [inline modules/spacecraftinstruments/rendering/renderablefov.cpp:276]: Should be `_standOffDistance = p.standOffDistance.value_or(_standOffDistance);`

File: modules/spacecraftinstruments/rendering/renderableplanetprojection.cpp [hunk 2]
```diff
@@ -322,25 +266,16 @@ RenderablePlanetProjection::RenderablePlanetProjection(const ghoul::Dictionary&
...
-    addProperty(_ambientBrightness);
-
+    
+    _maxProjectionsPerFrame = p.maxProjectionsPerFrame.value_or(_maxProjectionsPerFrame);
     addProperty(_maxProjectionsPerFrame);
+
+    addProperty(_ambientBrightness);
     addProperty(_projectionsInBuffer);
 
+
```
Comment (WeirdRubberDuck) [inline modules/spacecraftinstruments/rendering/renderableplanetprojection.cpp:278]: extra line

File: modules/sync/syncs/urlsynchronization.cpp [hunk 3]
```diff
@@ -46,105 +48,78 @@ namespace {
...
+        std::variant<std::string, std::vector<std::string>> url;
+
+        // This optional identifier will be part of the used folder structure and, if 
+        // provided, can be used to manually find the downloaded folder in the
+        // synchronization folder. If this value is not specified, 'UseHash' has to be set
+        // to 'true'
+        std::optional<std::string> identifier;
+
+        // If this value is set to 'true' and it is not overwritten by the global
+        //settings, the file(s) pointed to by this URLSynchronization will always be
```
Comment (WeirdRubberDuck) [inline modules/sync/syncs/urlsynchronization.cpp:64]: nitpicky, but missing whitespace after `//`

File: modules/sync/syncs/urlsynchronization.cpp [hunk 4]
```diff
@@ -46,105 +48,78 @@ namespace {
...
+        // provided, can be used to manually find the downloaded folder in the
+        // synchronization folder. If this value is not specified, 'UseHash' has to be set
+        // to 'true'
+        std::optional<std::string> identifier;
+
+        // If this value is set to 'true' and it is not overwritten by the global
+        //settings, the file(s) pointed to by this URLSynchronization will always be 
+        // downloaded, thus overwriting the local files. This is useful for files that are 
+        // updated regularly remotely and should be fetch at every startup
+        std::optional<bool> override;
```
Comment (WeirdRubberDuck) [inline modules/sync/syncs/urlsynchronization.cpp:67]: Not so nice that this variable has a name that corresponds to a C++ specifier. Don't know if it's actually going to be a problem though

Comment (WeirdRubberDuck) [reply modules/sync/syncs/urlsynchronization.cpp:67]: Maybe rename to `_forceOverride`, which is the name for the property, and use the `codegen::Key` thing?

File: modules/volume/tasks/generaterawvolumetask.cpp [hunk 5]
```diff
@@ -37,7 +37,7 @@
 #include <ghoul/lua/lua_helper.h>
 #include <ghoul/misc/dictionaryluaformatter.h>
 #include <ghoul/misc/defer.h>
-
+#include <ghoul/fmt.h>
```
Comment (WeirdRubberDuck) [inline modules/volume/tasks/generaterawvolumetask.cpp:40]: Move higher up in include list to preserve alphabetic sorting

File: modules/volume/tasks/generaterawvolumetask.cpp [hunk 6]
```diff
@@ -48,41 +48,64 @@ namespace {
...
+            "function taking three numbers as arguments (x, y, z) and returning a "
+            "number")]];
+
+        // The raw volume file to export data to
+        std::string rawVolumeOutput [[codegen::annotation("A valid filepath")]];
+
+        // The lua dictionary file to export metadata to
+        std::string dictionaryOutput [[codegen::annotation("A valid filepath")]];
+
+        std::string time;
```
Comment (WeirdRubberDuck) [inline modules/volume/tasks/generaterawvolumetask.cpp:64]: No documentation?

File: src/scene/lightsource.cpp [hunk 7]
```diff
@@ -32,17 +32,28 @@
...
-
     constexpr openspace::properties::Property::PropertyInfo EnabledInfo = {
         "Enabled",
         "Enabled",
         "Whether the light source is enabled or not"
     };
+
+    struct [[codegen::Dictionary(LightSource)]] Parameters {
+        // The type of the light source that is described in this element. The available
+        //types of light sources depend on the configuration of the application and can be
```
Comment (WeirdRubberDuck) [inline src/scene/lightsource.cpp:46]: nitpicky, butting missing whitespace after `//`

File: src/scripting/scriptscheduler_lua.inl [hunk 8]
```diff
@@ -38,7 +38,11 @@ int loadFile(lua_State* L) {
         return ghoul::lua::luaError(L, "filepath string is empty");
     }
 
-    ghoul::Dictionary scriptsDict = ghoul::lua::loadDictionaryFromFile(fileName, L);
+    ghoul::Dictionary scriptsDict;
+    scriptsDict.setValue("Scripts", ghoul::lua::loadDictionaryFromFile(fileName, L));
+
+
```
Comment (WeirdRubberDuck) [inline src/scripting/scriptscheduler_lua.inl:44]: Extra lines (also below the `testSpecificationAndThrow`). Intentional?

File: src/interaction/navigationhandler.cpp [hunk 9]
```diff
@@ -105,45 +131,29 @@ ghoul::Dictionary NavigationHandler::NavigationState::dictionary() const {
...
-    }
-    if (dictionary.hasValue<std::string>(KeyAim)) {
-        aim = dictionary.value<std::string>(KeyAim);
-    }
+    referenceFrame = p.referenceFrame.value_or(anchor);
+    aim = p.aim.value_or(aim);
 
-    if (dictionary.hasValue<glm::dvec3>(KeyUp)) {
-        up = dictionary.value<glm::dvec3>(KeyUp);
+    // @TODO (abock, 2021-03-24) This seems a bit odd;  is it correct that we only want to
```
Comment (WeirdRubberDuck) [inline src/interaction/navigationhandler.cpp:142]: Not completely sure, but I think the idea here is that the yaw and pitch angles does not really make sense unless you also know the up vector. Without an up vector we only know about the forward direction of the camera, and simply don't have sufficient information to compute the camera orientation from the yaw/pith. 

That being said, maybe the up-vector should actually just have a default value

**Review by WeirdRubberDuck — COMMENTED** [review]
👍

File: modules/sync/syncs/urlsynchronization.cpp (hunk 4 above)
Comment (alexanderbock) [reply modules/sync/syncs/urlsynchronization.cpp:67]: Yea;  good point.  It's not a problem compilation wise, but it confuses the syntax highlighter.

File: src/interaction/navigationhandler.cpp (hunk 9 above)
Comment (alexanderbock) [reply src/interaction/navigationhandler.cpp:142]: Didn't think about that.. i think it makes sense

**Pull request merged**